### ISSUE TITLE
The conversation keeps showing 'thinking...' and no reply for a long time.

### ISSUE THREAD

**Issue by TJtangjun** [microsoft/vscode-copilot-release#205]
When I ask a question on the side panel, Copilot keeps showing thinking and no reply for a long time.

 I checked the output screen and it shows: `message 0 returned. finish reason: [stop]`

![image](https://github.com/microsoft/vscode-copilot-release/assets/23148245/d1f2310b-a383-4ccd-99dc-bb90b80abce5)

**Comment by TJtangjun** [comment]
for example, the output:
```
[INFO] [default] [2023-06-05T02:39:56.144Z] [fetchChat] engine https://copilot-proxy.githubusercontent.com/v1/chat
[INFO] [default] [2023-06-05T02:39:57.717Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/chat/completions] took 1572 ms
[INFO] [streamMessages] [2023-06-05T02:39:58.198Z] message 0 returned. finish reason: [stop]
[INFO] [streamChoices] [2023-06-05T02:39:58.199Z] request done: headerRequestId: [2978c058-798d-432f-80dd-dedfca3276cf] model deployment ID: [xa68ad36c6722]
```

**Comment by Liaowag** [comment]
I'm having the same problem, always thinking. Trying to restart the conversation may help, but that doesn't always work either.

**Comment by Liaowag** [comment]
I tried changing the network and turning off any proxies related to github and it seems to have solved it for now.

**Comment by TJtangjun** [comment]
> I tried changing the network and turning off any proxies related to github and it seems to have solved it for now. This solution didn't work for me. I don't think it's a network problem, because the chat panel is the only one that has this problem, and the conversations in the editor are usually normal.

**Comment by Liaowag** [comment]
> > I tried changing the network and turning off any proxies related to github and it seems to have solved it for now.
> 
> This solution didn't work for me. I don't think it's a network problem, because the chat panel is the only one that has this problem, and the conversations in the editor are usually normal.

Indeed, you are right. Now the problem has come up again and it doesn't seem to be a network problem.

**Comment by davecheney** [comment]
`2978c058-798d-432f-80dd-dedfca3276cf`, the model produced 8 tokens then encountered the client provided stop condition `;`.

**Comment by unravel11** [comment]
I have same problem

**Comment by digitarald** [comment]
Per https://github.com/microsoft/vscode-copilot-release/issues/205#issuecomment-1579883769 , @lramos15 could this be related to meta prompting?

**Comment by TJtangjun** [comment]
I found that sometimes it does return the message, but it takes a long time to wait, like 5 minutes, or even longer. When you give up hope, go do something else, and come back, it surprisingly returns the correct reply. You don't even know when it replied.

**Comment by davecheney** [comment]
> I found that sometimes it does return the message, but it takes a long time to wait, like 5 minutes, or even longer. When you give up hope, go do something else, and come back, it surprisingly returns the correct reply. You don't even know when it replied.

Waiting longer than 60 second will not return a result, the chat model has a hard timeout at 60 seconds.

What could be happening is if you wait long enough without activity the underlying HTTP/2 connection will be closed -- somewhere between 5 and 15 minutes, it's not well characterised -- and that may "release" whatever inside the client is blocked waiting for input.

**Comment by kj415j45** [comment]
try ask the question after a greeting (like `Are you ready?`) solve my similar problem.

**Comment by TJtangjun** [comment]
> try ask the question after a greeting (like `Are you ready?`) solve my similar problem.

 Maybe I was too rude, it ignored me  >_<!

**Comment by TJtangjun** [comment]
Maybe if the first sentence is not in English, it will create some bugs.

**Comment by Tomahawk786** [comment]
I'm having the same problem

**Comment by Tongzhou-Yu** [comment]
> try ask the question after a greeting (like `Are you ready?`) solve my similar problem.

It's interesting that works. Can't understand

**Comment by kj415j45** [comment]
> I think we know where the bug is here, but just to confirm. Do you only see this on the first use of chat and only on occasion? Or does it happen multiple times in a session?

@lramos15 It usually happened at the beginning of a chat conversation. If it succeed to respond the first message, following conversation is unlikely to make copilot keep thinking.

**Comment by predator17** [comment]
i encountered same issue and was able to fix via switching to github account in vs code rather than microsoft account, 
hopefully this would help.

**Comment by Cupkiller-vip** [comment]
Unfortunately, I'm also experiencing this problem and the output is the same as yours. 

I don't think it's generated due to network issues as GitHub Copilot is working and Fix using copilot is available. I set up a proxy in vscode and it doesn't work. I tried all of the above, however none of them worked. It happens in every chat, in fact I haven't managed to get a single reply yet.

**Comment by Cupkiller-vip** [comment]
> Does inline chat (via CTRL / CMD + I) work?

This won't work because when you type /explain in the input box, you'll realize that it's actually still executing on the left side. However, today it changed in some confusing way - it works properly. What caused this issue to appear and be fixed?

**Comment by lramos15** [comment]
> > Does inline chat (via CTRL / CMD + I) work?
> 
> This won't work because when you type /explain in the input box, you'll realize that it's actually still executing on the left side. However, today it changed in some confusing way - it works properly. What caused this issue to appear and be fixed?

There has been no changes 🤷

**Comment by harveydobson** [comment]
I had this same issue just now with it responding to every prompt with "Thinking..." and it was because the file that the IDE was focussed on has 51,000 lines which I would imagine is too large for the Github Copilot context. 

Once I realised this and switched to a different file, it was fine.
